### ISSUE TITLE
Weird error message when setting class variable of the wrong type (not in an instance)

### ISSUE THREAD

**Issue by svenzorgdoc** [microsoft/pyright#6106]
**Describe the bug**
There is something weird with Pyright's error message when trying to set an unannotated class variable on the class itself when using an incorrect type.

**Code or Screenshots**

```python
from typing import ClassVar


class Test:
    number = 1
    text = "hello"
    number_as_classvar: ClassVar[int] = 1
    text_as_classvar: ClassVar[str] = "hello"


Test.number = 4  # no error, correct
Test.text = 4  # WEIRD ERROR

Test.number_as_classvar = 4  # no error, correct
Test.text_as_classvar = 4  # correct error

Test.fdjkdfs = 3  # correct error
```

The result is:

```
pyright 1.1.300
/tmp/huh.py
  /tmp/huh.py:12:6 - error: Cannot assign member "text" for type "Type[Test]"
    Member "text" is unknown (reportGeneralTypeIssues)
  /tmp/huh.py:15:25 - error: Cannot assign member "text_as_classvar" for type "Type[Test]"
    "Literal[4]" is incompatible with "str" (reportGeneralTypeIssues)
  /tmp/huh.py:17:6 - error: Cannot assign member "fdjkdfs" for type "Type[Test]"
    Member "fdjkdfs" is unknown (reportGeneralTypeIssues)
3 errors, 0 warnings, 0 informations 
```

The error on `/tmp/huh.py:12:6` is unexpected.

**VS Code extension or command-line**
Tested using pyright command line tool version 1.1.300.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.331, which I just published. It will also be included in a future release of pylance.
